public/components/integrations/components/setup_integration.tsx

```diff
@@ -148,95 +177,225 @@ const addIntegration = async ({
+  }
+};
+
+/**
+ * Handles the installation process for S3 integration by creating a database (if specified),
+ * processing integration assets, and executing necessary queries.
+ *
+ * @param {Object} params - The parameters object
+ * @param {IntegrationSetupInputs} params.config - Configuration settings for the integration setup
+ * @param {IntegrationConfig} params.integration - Integration configuration details
+ * @param {Function} params.setLoading - Callback function to set loading state
+ * @param {Function} params.setCalloutLikeToast - Callback function to display toast notifications
+ * @param {string} [params.dataSourceMDSId] - Optional MDS ID for the data source
+ * @param {string} [params.dataSourceMDSLabel] - Optional MDS label for the data source
+ * @param {Function} [params.setIsInstalling] - Optional callback to set installation status
+ *
+ * @returns {Promise<void>} A promise that resolves when the installation is complete
+ *
+ * @throws Will set error toast if database creation fails or integration addition fails
+ */
+const addFlintIntegration = async ({
+  config,
+  integration,
+  setLoading,
+  setCalloutLikeToast,
+  dataSourceMDSId,
+  dataSourceMDSLabel,
+  setIsInstalling,
+}: {
+  config: IntegrationSetupInputs;
+  integration: IntegrationConfig;
+  setLoading: (loading: boolean) => void;
+  setCalloutLikeToast: (title: string, color?: Color, text?: string) => void;
+  dataSourceMDSId?: string;
+  dataSourceMDSLabel?: string;
+  setIsInstalling?: (isInstalling: boolean, success?: boolean) => void;
+}): Promise<void> => {
+  let sessionId: string | undefined;
+
+  // Create database if specified
+  if (config.databaseName) {
+    const createDbQuery = `CREATE DATABASE IF NOT EXISTS ${config.databaseName}`;
+    const result = await runQuery(
+      createDbQuery,
+      config.connectionDataSource,
+      sessionId,
+      dataSourceMDSId
+    );
+
+    if (!result.ok) {
+      setLoading(false);
+      setCalloutLikeToast('Failed to create database', 'danger', result.error.message);
+      return;
+    }
+    sessionId = result.value.sessionId;
+  }
+
+  // Process integration assets
+  const http = coreRefs.http!;
+  const assets: { data: ParsedIntegrationAsset[] } = await http.get(
+    `${INTEGRATIONS_BASE}/repository/${integration.name}/assets`
+  );
+
+  // Execute queries
+  for (const query of assets.data.filter(
+    (a: ParsedIntegrationAsset): a is ParsedIntegrationAsset & { type: 'query' } =>
+      a.type === 'query'
+  )) {
+    if (query.workflows && !query.workflows.some((w) => config.enabledWorkflows.includes(w))) {
+      continue;
     }
-    if (!res) {
+
+    const queryStr = prepareQuery(query.query, config);
+    const result = await runQuery(
+      queryStr,
+      config.connectionDataSource,
+      sessionId,
+      dataSourceMDSId
+    );
+
+    if (!result.ok) {
       setLoading(false);
+      setCalloutLikeToast('Failed to add integration', 'danger', result.error.message);
```

Comment:
thought: At some point we need to register all these strings to i18n, but do we need to do that now?

@ps48 / @TackAdam any context on i18n timelines, or is there a doc that describes the migration process? I can make an issue to do it later.

Reply:
Not blocking, is good to have but repo needs a pass to be uniform in its implementation.